Sources/Profile-Modify.php

```diff
@@ -2253,20 +2253,45 @@ function alert_delete($toDelete, $memID = false)
 
 /**
  * Counts how many alerts a user has - either unread or all depending on $unread
+ * We can't use db_num_rows here, as we have to determine what boards the user can see
+ * Possibly in future versions as database support for json is mainstream, we can simplify this.
  *
  * @param int $memID The user ID.
  * @param bool $unread Whether to only count unread alerts.
  * @return int The number of requested alerts
  */
 function alert_count($memID, $unread = false)
 {
-	global $smcFunc;
+	global $smcFunc, $user_info;
 
 	if (empty($memID))
 		return false;
 
+	// Leave a place holder here for now.
+	// MySQL uses JSON_CONTAINS() to find data, Postgresql seems to be able to natively use it with JSONB column type.
+	// Maybe a $smcFunc['db_json_supported']?
+	if (false)
```

Comment:
How would look the json approached?

Reply:
It has been so long, I honestly am not 100% sure, but the notes say that postgres and mysql handled json differently.

I don't think its a case for something in 2.1 at this point.  Maybe a future version we can identify a way to natively handle json string searches.